### PR TITLE
chore: Update demo-app to use functional component syntax

### PR DESCRIPTION
**Pull request by ggdouglas**
#### Checklist

- [ ] Includes tests
- [ ] Update documentation

### CODE REVIEW

**Review by bvandrc — DISMISSED** [review]

File: packages/demo-app/src/examples/ExampleCard.tsx [hunk 1]
```diff
@@ -21,29 +21,28 @@ import { Card, H5 } from "@blueprintjs/core";
...
+    return (
+        <div className="example-card-container">
+            <H5>{label}</H5>
+            {subLabel && <H5>{subLabel}</H5>}
+            <Card className={classNames("example-card", { horizontal })} elevation={0} style={{ width }}>
+                {children}
+            </Card>
+        </div>
+    );
+};
```
Comment (invliD) [inline packages/demo-app/src/examples/ExampleCard.tsx:48]: Add a display name?

File: packages/demo-app/src/examples/Examples.tsx [hunk 2]
```diff
@@ -47,47 +47,46 @@ import { ToastExample } from "./ToastExample";
...
-        );
-    }
+export const Examples: React.FC = () => {
+    return (
+        <div className="examples-root">
+            <ExamplesContainer />
+            <ExamplesContainer isDark={true} />
+        </div>
+    );
+};
```
Comment (invliD) [inline packages/demo-app/src/examples/Examples.tsx:57]: same here

File: packages/demo-app/src/examples/Examples.tsx [hunk 3]
```diff
@@ -47,47 +47,46 @@ import { ToastExample } from "./ToastExample";
...
+            <TabsExample />
+            <TagExample />
+            <TagInputExample />
+            <TextExample />
+            <ToastExample />
+            <TooltipExample />
+            <TreeExample />
+        </div>
+    );
+};
```
Comment (invliD) [inline packages/demo-app/src/examples/Examples.tsx:94]: and here

**Review by invliD — DISMISSED** [review]

File: packages/demo-app/src/examples/ExampleCard.tsx (hunk 1 above)
Comment (ggdouglas) [reply packages/demo-app/src/examples/ExampleCard.tsx:48]: 2adcb400993aec04991905b8b7d394a64b410768

**Pull request merged**